www/utils/retryUtils.ts

```diff
@@ -0,0 +1,72 @@
+import retry from "retry";
+import { captureException, captureMessage } from "@sentry/nextjs";
+
+interface RetryOptions {
+  retries: number;
+  factor: number;
+  minTimeout: number;
+  maxTimeout: number;
+}
+
+const dbOptions: RetryOptions = {
+  retries: 3,
+  factor: 1.5,
+  minTimeout: 1000,
+  maxTimeout: 10000,
+};
+
+const openAIOptions: RetryOptions = {
+  retries: 5,
+  factor: 2,
+  minTimeout: 4000,
+  maxTimeout: 60000,
+};
```

Comment:
What are we basing these numbers on just curious and why they are different from the dbOptions one.

Reply:
The thinking is that, were we to hit a rate limit error, we could potentially space out the retries to hit a new window for token limits (I believe they are 60 sec). Without actually testing it's tough to say if this would actually work in practice.